src/EncounterPage/HPCalculator.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react'
+import CalculatorButton from './CalculatorButton'
+import styled from 'styled-components'
+
+export default function HPCalculator({ setCurrentHP, currentHP, maxHP }) {
+  const [hpInputValue, setHpInputValue] = useState('')
+  return (
+    <CounterStyled>
+      <CalculatorButton onClick={addToHP} operator="+" />
+      <InputStyled
+        min={0}
+        value={hpInputValue}
+        onChange={(event) => handleChange(event)}
+        name="HPInput"
+        type="tel"
+      />
+
+      <CalculatorButton onClick={subtractFromHP} operator="-" />
+    </CounterStyled>
+  )
+  function handleChange(event) {
+    const hpInputValue = event.target.value
+    const regexp = new RegExp('^[0-9]+$')
+    if (regexp.test(hpInputValue)) {
+      setHpInputValue(hpInputValue)
+    }
+  }
+
+  function addToHP() {
+    const calculatedHP = currentHP + Number(hpInputValue)
+    setCurrentHP(calculatedHP > maxHP ? maxHP : calculatedHP)
+    setHpInputValue('')
+  }
+  function subtractFromHP() {
+    const calculatedHP = currentHP - Number(hpInputValue)
+    setCurrentHP(calculatedHP < 0 ? 0 : calculatedHP)
+    setHpInputValue('')
+  }
```

Comment:
Hier könntest du die Abfrage einfügen, dass die currentHP  > als die hpInputValue sein muss, bzw. mindestens 1, sov erhinderst du, dass das Ergebnis negativ sein kann :)

Reply:
Die Logik funktioniert bereits so, dass currentHP niemals unter 0 werden kann, weil der State automatisch auf 0 gesetzt wird sobald es negativ werden würde. Ich hatte nur in der EnemyEntry Komponente vergessen meine alte unschöne Lösung ```HP: {currentHP > 0 ? currentHP : 0}/{maxHP}``` durch ```HP: {currentHP}/{maxHP}``` zu ersetzen. Das wirkte jetzt natürlich etwas verwirrend :D Trotzdem danke für den Hinweis :)